snorkel/labeling/model/label_model.py

```diff
@@ -545,6 +545,16 @@ def _set_class_balance(
             self.p = sorted_counts / sum(sorted_counts)
         else:
             self.p = (1 / self.cardinality) * np.ones(self.cardinality)
+
+        if np.any(self.p == 0):
+            raise ValueError(
+                f"Class balance prior is 0 for class(es) {np.where(self.p)[0]}."
+            )
+        if len(self.p) != self.cardinality:
+            raise ValueError(
+                f"Y_dev has {len(self.p)} class(es). Does not match LabelModel cardinality {self.cardinality}."
```

Comment:
this could also be because `class_balance` is the wrong size

Reply:
this error is also confusing for the Y_dev case. what's happening is that we're saying some of the class priors are 0 (by virtue of not being in Y_dev) which is the same as the above ValueError. rec: we raise a separate message under each of the if/else block (under `if class_balance is not None:` if it's the wrong size, then something like`Class balance prior is 0` in the other two cases.